mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -98,8 +98,17 @@ static int StringLength (string a)
 			return a?.Length ?? 0;
 		}
 
+		static void CheckString (string name, string value)
+		{
+			// Native code tends to truncate at NUL which is incorrect. Guard it here.
+			if (value?.IndexOf((char)0) >= 0)
+				throw new ArgumentException ("String must not contain embedded NULs.", name);
+		}
+
 		internal static unsafe IntPtr OpenFile (string path, FileMode mode, string mapName, out long capacity, MemoryMappedFileAccess access, MemoryMappedFileOptions options)
 		{
+			CheckString ("path", path);
+			CheckString ("mapName", mapName);
```

Comment:
these should ideally use `nameof (path)` for the first argument but I'm fine avoiding yet another iteration in the name of making progress ;)